test/e2e/client_tls_test.go

```diff
@@ -384,6 +313,535 @@ func TestClientTLS(t *testing.T) {
+				}
+				signedByRoot.CertPem = strings.Join([]string{signedByRoot.CertPem, rootCA.CertPem}, "\n")
+				signedByIntermediate, err := CreateTLSKeyCert("client signed by intermediate", time.Now(), time.Now().Add(24*time.Hour), false, nil, &intermediateCA)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+				signedByIntermediate.CertPem = strings.Join([]string{signedByIntermediate.CertPem, intermediateCA.CertPem, rootCA.CertPem}, "\n")
+				revokedByRoot, err := CreateTLSKeyCert("client revoked by root", time.Now(), time.Now().Add(24*time.Hour), false, nil, &rootCA)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+				revokedByRoot.CertPem = strings.Join([]string{revokedByRoot.CertPem, rootCA.CertPem}, "\n")
+				revokedByIntermediate, err := CreateTLSKeyCert("client revoked by intermediate", time.Now(), time.Now().Add(24*time.Hour), false, nil, &intermediateCA)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+				revokedByIntermediate.CertPem = strings.Join([]string{revokedByIntermediate.CertPem, intermediateCA.CertPem, rootCA.CertPem}, "\n")
+				selfSigned, err := CreateTLSKeyCert("self signed cert", time.Now(), time.Now().Add(24*time.Hour), false, nil, nil)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+
+				_, rootCRLPem, err := CreateCRL(nil, rootCA, time.Now(), time.Now().Add(10*time.Minute), RevokeCertificates(time.Now(), revokedByRoot))
+				if err != nil {
+					t.Fatalf("Failed to create root CRL: %v", err)
+				}
+				_, intermediateCRLPem, err := CreateCRL(nil, intermediateCA, time.Now(), time.Now().Add(10*time.Minute), RevokeCertificates(time.Now(), revokedByIntermediate))
+				if err != nil {
+					t.Fatalf("Failed to create intermediate CRL: %v", err)
+				}
+
+				return TestMTLSWithCRLsCerts{
+					CABundle: []string{
+						intermediateCA.CertPem,
+						rootCA.CertPem,
+					},
+					CRLs: map[string]string{
+						"root-ca.crl":         rootCRLPem,
+						"intermediate-ca.crl": intermediateCRLPem,
+					},
+					ClientCerts: struct {
+						Accepted map[string]KeyCert
+						Rejected map[string]KeyCert
+					}{
+						Accepted: map[string]KeyCert{
+							"signed-by-root":         signedByRoot,
+							"signed-by-intermediate": signedByIntermediate,
+						},
+						Rejected: map[string]KeyCert{
+							"revoked-by-root":         revokedByRoot,
+							"revoked-by-intermediate": revokedByIntermediate,
+							"self-signed":             selfSigned,
+						},
+					},
+				}
+			},
+		},
+		{
+			// This test case has certificates including the CRL distribution point of their signer (i.e. intermediate
+			// CA is signed by root CA, and includes the URL for root's CRL). In this case, neither of the certificates
+			// in the CA bundle include the intermediate CRL, so connections that rely on it will be rejected.
+			Name: "signer-crl",
```

Comment:
I understand this name better, but could be more explicit `crls-included-in-signer-certs` or something.